lib/spack/spack/build_systems/cmake.py

```diff
@@ -255,10 +255,12 @@ def check(self):
         """
         with working_dir(self.build_directory):
             if self.generator == 'Unix Makefiles':
-                self._if_make_target_execute('test')
+                self._if_make_target_execute('test',
+                                             jobs_env='CTEST_PARALLEL_LEVEL')
```

Comment:
how does one use it? Could you please add a sentence in class documentation?

there is a dedicated PR on documentation for build systems https://github.com/spack/spack/pull/5015/ but it is not merged yet.

Reply:
@davydden If you're talking specifically about the `cmake.py` change you noted above, "one" doesn't "use" it. It's a change in the implementation to provide the parallelism that one could reasonably expect for the `test` target in the event that it is CMake-generated (_i.e._ invokes `ctest`). This implementation _uses_ the expanded functionality of `MakeExecutable`, which is actually explained in the class documentation therefor.
In your case, where the `test` target is hand-provided by your package rather than generated by CMake, there will be no effect _unless_ (of course) your target invokes `ctest`.

Reply:
@davydden I'm happy that it does what it set out to do. If everyone else thinks it's a reasonable implementation, I'm happy for it to be merged.